internal/controllers/orano2ims_controller_test.go

```diff
@@ -0,0 +1,348 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	oranv1alpha1 "github.com/openshift-kni/oran-o2ims/api/v1alpha1"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	"github.com/stretchr/testify/assert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var (
+	testscheme = scheme.Scheme
+)
+
+func init() {
+	testscheme.AddKnownTypes(oranv1alpha1.GroupVersion, &oranv1alpha1.ORANO2IMS{})
+	testscheme.AddKnownTypes(oranv1alpha1.GroupVersion, &oranv1alpha1.ORANO2IMSList{})
+	testscheme.AddKnownTypes(networkingv1.SchemeGroupVersion, &networkingv1.Ingress{})
+	testscheme.AddKnownTypes(networkingv1.SchemeGroupVersion, &networkingv1.IngressList{})
+	testscheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.ServiceAccount{})
+	testscheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.ServiceAccountList{})
+	testscheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.Service{})
+	testscheme.AddKnownTypes(corev1.SchemeGroupVersion, &corev1.ServiceList{})
+	testscheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.Deployment{})
+	testscheme.AddKnownTypes(appsv1.SchemeGroupVersion, &appsv1.DeploymentList{})
+}
```

Comment:
Is there any chance that we can use the Ginkgo framework for this tests like we do for the rest of the tests of the project? For example, for instead of this `init` we could do something like this:

```go
var _ = BeforeAll(func () {
        testscheme.AddKnownTypes(...)
})